### PR TITLE
wired updated operator framework

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/1839. Depends on https://github.com/giantswarm/operatorkit/pull/59.

### CODE REVIEW

File: glide.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ import:
 - package: github.com/giantswarm/microkit
   version: master
 - package: github.com/giantswarm/operatorkit
-  version: master
+  version: update
```
Comment (xh3b4sd) [inline glide.yaml]: TODO

File: service/operator/service.go [hunk 2]
```diff
@@ -141,10 +133,8 @@ func (s *Service) bootWithError() error {
 
 	s.logger.Log("debug", "starting list/watch")
 
-	newResourceEventHandler := &cache.ResourceEventHandlerFuncs{
-		AddFunc:    s.addFunc,
-		DeleteFunc: s.deleteFunc,
-	}
+	newResourceEventHandler := s.operatorFramework.NewCacheResourceEventHandler()
```
Comment (xh3b4sd) [inline service/operator/service.go]: The operators are now less complex, which I like.

File: service/service.go [hunk 3]
```diff
@@ -61,11 +61,21 @@ func DefaultConfig() Config {
...
 
+type Service struct {
+	// Dependencies.
+	Healthz  *healthz.Service
+	Operator *operator.Service
+	Version  *version.Service
+
+	// Internals.
+	bootOnce sync.Once
+}
```
Comment (xh3b4sd) [inline service/service.go]: Just boyscouting here.

File: service/service.go [hunk 4]
```diff
@@ -61,11 +61,21 @@ func DefaultConfig() Config {
...
+
 // New creates a new configured service object.
 func New(config Config) (*Service, error) {
 	// Dependencies.
 	if config.Logger == nil {
-		return nil, microerror.Maskf(invalidConfigError, "logger must not be empty")
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
```
Comment (xh3b4sd) [inline service/service.go]: Just boyscouting here.

File: service/service.go [hunk 5]
```diff
@@ -177,11 +187,26 @@ func New(config Config) (*Service, error) {
...
 		frameworkConfig.Logger = config.Logger
+		frameworkConfig.Resources = []framework.Resource{
+			namespaceResource,
+
+			configMapResource,
+			deploymentResource,
+			ingressResource,
+			pvcResource,
+			serviceResource,
+		}
```
Comment (xh3b4sd) [inline service/service.go]: The framework itself is now configured with the resources, not the specific operator service. Now we can use the common cache resource event handler. I think it is better that way.

File: glide.yaml (hunk 1 above)
Comment (xh3b4sd) [reply glide.yaml]: Done.

File: service/operator/service.go [hunk 6]
```diff
@@ -154,37 +144,3 @@ func (s *Service) bootWithError() error {
...
-	s.mutex.Lock()
-	defer s.mutex.Unlock()
-
-	s.logger.Log("debug", "executing the operator's deleteFunc")
-
-	err := s.operatorFramework.ProcessDelete(obj, s.resources)
-	if err != nil {
-		s.logger.Log("error", fmt.Sprintf("%#v", err), "event", "delete")
-	}
-}
```
Comment (kopiczko) [inline service/operator/service.go]: This change is very good. Gives us a lot of flexibility. <3

**Pull request merged**